uc.c

```diff
@@ -1144,6 +1144,34 @@ uint32_t uc_mem_regions(uc_engine *uc, uc_mem_region **regions, uint32_t *count)
 }
 
 UNICORN_EXPORT
+uc_err uc_mmio_map(uc_engine *uc, uint64_t address, size_t size,
+        uc_cb_mmio_t callback, void *user_data)
+{
+    uc_err res;
+    struct mmio_data *data;
+
+    if (callback == NULL)
+        return UC_ERR_ARG;
+
+    if (uc->mem_redirect) {
+        address = uc->mem_redirect(address);
+    }
+
+    res = mem_map_check(uc, address, size, UC_PROT_ALL);
+    if (res)
+        return res;
+
+    data = g_new(struct mmio_data, 1);
```

Comment:
... DOH. Can't believe that I forgot about this. It *is* leaking at the moment, isn't it? Need to: (A) remove the need for that struct. (B) be able to tell mmio apart from ram and free that memory when unmapping the mmio region. (C) create a seperate unmapping function.

You have any ideas on how to accomplish A? I don't mind changing the API to require 2 callbacks, for example, just depends on what you think is the best interface to present. I'll be happy that the API is there.